### ISSUE TITLE
Objects are not valid as a React child (found: object with keys {title})

### ISSUE THREAD

**Issue by PCreations** [cht8687/react-listview-sticky-header#6]
From the example, (with react 0.14)

Invariant Violation: Objects are not valid as a React child (found: object with keys {title}). If you meant to render a collection of children, use an array instead or wrap the object using createFragment(object) from the React add-ons. Check the render method of `ListItem`.

**Comment by cht8687** [comment]
Hi @PCreations , it appears that in 0.14, React doesn't accept object as children anymore. I will further investigate this tonight. https://github.com/facebook/react/issues/5139

**Comment by cht8687** [comment]
Hi @PCreations , this had been fixed in a77e1f58124a2bbca0036fca0ac7f0308e052ee5. Thanks.

**Comment by HubMahesh** [comment]
Hi there i am working with react native.
and want list using map function. not by list item given by default in react js. Because i have to show more data in a box list.

my json data as follows 


data[{
  "status": "FOUND",
  "data": [
    {
      "ID": 7496630809,
      "details": "North coria",
      "amount": "700",
      "issued_date": "01/02/2016",
      "type": "personal",
      "allow": true,
    },
    {
      "ID": 789767595,
      "details": "uthan",
      "amount": "700",
      "issued_date": "01/07/2011",
      "type": "public",
      "allow": true,
    },
    {
     "ID": 6587568,
      "details": "monica",
      "amount": "86.99",
      "issued_date": "23/08/2011",
      "type": "personal",
      "allow": true,
    },
    {
      "ID": 86867,
      "details": "Sappan",
      "amount": "4800",
      "issued_date": "15/07/2010",
      "type": "personal",
      "allow": true,
    }
  ]


My code is

{this.state.data.map((item,i) => <Text key={i}>{item} </Text>)}

every time showing error "Object are not valid as a React Child (Found object with keys).

So, can anyone facing this problem.

**Comment by SiddharthaSarma** [comment]
Don't pass an object as a prop to React.child. Instead, pass it as { ... item} and then access using props.{property} that might fix your problem

**Comment by carendivinagracia22** [comment]
@SiddharthaSarma this solves my problem :100:

**Comment by MrShakes** [comment]
Please someone expatiate by providing sample code that shows @SiddharthaSarma solution in action, I'm doing something similar to this:
`{this.state.data.map((item, i) => {item} )}`

**Comment by Cre3z** [comment]
@MrShakes something like this:
`{
                this.state.data.map((field, index) => {
                    return <Yourcomponent {... field}/>
                })
            }`

**Comment by umairkhan1997** [comment]
i am just importing firebase in react native  , it gave me an error :
 "Objects are not valid as a React child (found: object with keys {$$typeof, type, key, ref, props, _owner, _store}). If you meant to render a collection of children, use an array instead."

**Comment by prakashnov07** [comment]
I had the same problem , I placed the import statement for firebase in the end of all the imports. It worked for me.

**Comment by bravechou2009** [comment]
### 

> 

- [ ] 

- [ ]   #@`_**
Duplicate of #**_`

**Comment by th3knigh7** [comment]
if anyone is still having this problem, remove firebase import statement and add it in one of the lifecycle methods as below.

before:
`import React, { Component } from 'react';
import { Text, View } from 'react-native';
import firebase from 'firebase';    /*Remove this line*/ 

class App extends Component {
    componentWillMount(){
     /*put it here instead*/
}`

After:
`import React, { Component } from 'react';
import { Text, View } from 'react-native';

class App extends Component {
    componentWillMount(){
          const firebase = require("firebase");
}`

**Comment by sathiyaanbu** [comment]
I have the same issue but still couldn't get all the data in to table. Get only one row because of results[0] . Any suggestions pls?

```import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchData } from '../actions/index';

import { Table, img } from 'reactstrap';

class DataList extends Component {
  componentWillMount() {
    this.props.fetchData();
  }

  render() {
    const dataItems = this.props.results.map(result => (
      <tr key={result.results[0].id}>
        <td>{result.results[0].id}</td>
        <td>
          <img
            style={{ width: '150px', height: '100px' }}
            src={result.results[0].image}
          />
        </td>
        <td>{result.results[0].name}</td>
        <td>{result.results[0].status}</td>
        <td>{result.results[0].species}</td>
      </tr>
    ));

    return (
      <Table bordered>
        <thead>
          <tr>
            <th>id</th>
            <th>Character image</th>
            <th>Name</th>
            <th>Status</th>
            <th>Species</th>
          </tr>
        </thead>
        <tbody>{dataItems}</tbody>
      </Table>
    );
  }
}

const mapStateToProps = state => ({
  results: state.results
});

export default connect(mapStateToProps, { fetchData })(DataList);
```

**Comment by abdekp** [comment]
I have the same issue as umairkhan1997, I tried th3knigh7's answer but did not work. Is it related to firebase itself or something with npm?

**Comment by th3knigh7** [comment]
@abdekp 
problem is related to the latest version of the firebase. if the solution above did not work for you try installing the previous version of the firebase.  like this:
`npm install firebase@5.0.3`

**Comment by afoysal** [comment]
@Cre3z When should I use `{ this.state.data.map((item, index) => { return <Yourcomponent {... item}/> }) }` and when should I use `{ this.state.data.map((item, index) => { return <Yourcomponent element={... item}/> }) }` ?

**Comment by patiljignesh** [comment]
Thank you @th3knigh7 !!

**Comment by usman1579** [comment]
th3knigh7 thanks it work for me..you save my time.

**Comment by whizdummy** [comment]
@Nisharunnisaa Try to upgrade your dependency to at least React 16.1.1. The library you used depends on v16.1.1.

**Comment by Nisharunnisaa** [comment]
@whizdummy  the following are the libraries i used:
`  "expo": "^27.0.1",
    "firebase": "^5.3.0",
    "react": "16.3.1",
    "react-native": "~0.55.2",
    "react-native-card-view": "0.0.3",
    "react-native-charts-wrapper": "^0.4.4",
    "react-native-elements": "^0.19.1",
    "react-native-pie": "^0.5.0",
    "react-native-radio-button": "^2.0.1",
    "react-native-simple-card-view": "^0.3.0",
    "react-native-simple-radio-button": "^2.7.2",
    "react-native-svg-charts": "^5.2.0",
    "react-native-table-component": "^1.1.8",
    "react-navigation": "^2.9.2",
    "react-navigation-material-bottom-tabs": "^0.3.0"`

**Comment by alexandreseabra** [comment]
Same error message from @Nisharunnisaa . Even when using react  v16.5.0

**Comment by usman1579** [comment]
Hi! let me tell you the solution ...

**Comment by Harsh4861** [comment]
> @abdekp
> problem is related to the latest version of the firebase. if the solution above did not work for you try installing the previous version of the firebase. like this:
> `npm install firebase@5.0.3`

this work for me ,thanks

**Comment by esteechenyiwei** [comment]
same problem as @alexandreseabra  and @Nisharunnisaa , using React 16.12....

**Comment by swatiwak** [comment]
facing same issue 
Uncaught Error: Objects are not valid as a React child (found: [object HTMLDivElement]). If you meant to render a collection of children, use an array instead.

Does anyone know solution for this.

**Comment by devbysan** [comment]
setPosts(snapshot.docs.map(doc =>
      ({
        id: doc.id,
        post: doc.data()
      })
      ))

**Comment by saurabhkanswal** [comment]
@devbysan  hey, did you get a solution?

**Comment by THEORIGINITE** [comment]
@devbysan  IF YOU GOT THE SOLUTION PLEASE LET ME KNOW PLEASE

**Comment by execptionerror** [comment]
[[Solved] Objects are not valid as a React child. If you meant to render a collection of children, use an array instead](https://exerror.com/objects-are-not-valid-as-a-react-child-if-you-meant-to-render-a-collection-of-children-use-an-array-instead/)
